geospaas/nansat_ingestor/tests.py

```diff
@@ -112,10 +111,15 @@ def setUp(self):
         # in order to prevent "mock leak" in the tests
         self.addCleanup(self.patcher.stop)
 
+        self.patcher2 = patch('geospaas.utils.utils.os.path.isfile')
+        self.mock_isfile = self.patcher2.start()
+        self.mock_isfile.return_value = True
```

Comment:
We actually don't use `isfile directly` in `nansat_ingestor.managers.DatasetManager`. Instead we use `validate_uri`. 
```suggestion
        self.patcher2 = patch('geospaas.nansat_ingestor.managers.validate_uri')
        self.mock_validate_uri = self.patcher2.start()
        self.mock_validate_uri.return_value = None
```

Reply:
@akorosov 
We have a test that prevent us from patching the validate_uri. Because this test examine the correct behaviour of this function:
https://github.com/nansencenter/django-geo-spaas/blob/e333ebf98ed97da470e02400513d617793e1b373/geospaas/nansat_ingestor/tests.py#L177
What should I do? remove this test and patch the validate_uri? or patch the isfile?

Reply:
To me it seems that this test doesn't belong here. Can you move it to geospaas.tests.TestUtils ?

BTW, I noticed a typo in my suggestion. It should be self.addCleanup(patch.stopall) (instead of patcher.stopall).

Reply:
@akorosov 

Yeah, that typo take some minutes for me to find out the correct one. No problem. 


If I want to move it, then I have to import dataset from ingestor models in the TestUtils. Is it a good import for TestUtils?

Reply:
I don't quite undersnad how this test works, to be hones. I cannot find in the code where the ValueError should be raised. I don;t see that in the  DatasetManager so I assume that it happens inside validate_uri. Therefore in TestUtils it should not call Dataset.objects.get_or_create(invalid_uri), but validate_uri(invalid_uri).